nuxeo-core/nuxeo-core-storage-dbs/src/main/java/org/nuxeo/ecm/core/storage/dbs/DBSTransactionState.java

```diff
@@ -151,13 +161,24 @@
 
     protected final Set<String> browsePermissions;
 
-    public DBSTransactionState(DBSRepository repository, DBSSession session) {
+    public DBSTransactionState(DBSRepository repository, DBSConnection connection, DBSSession session) {
         this.repository = repository;
+        this.connection = connection;
         this.session = session;
         SecurityService securityService = Framework.getService(SecurityService.class);
         browsePermissions = new HashSet<>(Arrays.asList(securityService.getPermissionsToCheck(BROWSE)));
     }
 
+    /** @since 11.1 */
+    public void close() {
+        connection.close();
+    }
```

Comment:
Can we make DBSTransactionState a `Closeable`?

Reply:
Yes I don't think it's a problem.

Reply:
`Autocloseable` actually has this really has nothing to do with `java.io`